File: java/数组/1877. 数组中最大数对和的最小值.java
class Solution {
    public int minPairSum(int[] nums) {
        int n = nums.length;
        if (n <= 2) {
            return Arrays.stream(nums).sum();
        }
        Arrays.sort(nums);
        int max_sum = 0;

        for (int i = 0; i < n >> 1; i++) {
            max_sum = Math.max(max_sum, nums[i] + nums[n - 1 - i]);
        }

        return max_sum;
    }
}

File: python/数组/1877. 数组中最大数对和的最小值.py
class Solution:
    def minPairSum(self, nums: list[int]) -> int:
        n = len(nums)
        if n <= 2:
            return sum(nums)
        nums.sort()
        max_sum = float('-inf')

        """
        n = 4
        0, 3
        1, 2
        """
        for i in range(0, n >> 1):
            max_sum = max(max_sum, nums[i] + nums[n - 1 - i])
        
        return max_sum